its/tests/src/test/java/its/ConnectedModeTest.java

```diff
@@ -689,6 +680,21 @@ public void cleanOldPlugins() throws IOException {
     assertThat(newFile).doesNotExist();
   }
 
+  @Test
+  @Ignore
+  public void updatesStorageOnServerEvents() throws IOException, InterruptedException {
+    updateGlobal();
+    updateProject(PROJECT_KEY_JAVA);
+    engine.subscribeForEvents(endpointParams(ORCHESTRATOR), sqHttpClient(), Set.of(PROJECT_KEY_JAVA));
+    var qualityProfile = getQualityProfile("SonarLint IT Java");
+    deactivateRule(qualityProfile, "S106");
+    Thread.sleep(2000);
+
+    var issueListener = new SaveIssueListener();
+    engine.analyze(createAnalysisConfiguration(PROJECT_KEY_JAVA, PROJECT_KEY_JAVA, "src/main/java/foo/Foo.java"), issueListener, null, null);
```

Comment:
It would make the test more robust IMO if you were running the same analysis twice: once before deactivating the rule, and once after

Reply:
Why more robust ? There is already another test in this suite that checks this

Reply:
This is a personal taste (when it is not too costly) to have a full "scenario".